### PR TITLE
Get diagnostics

### PR DESCRIPTION
**Pull request by rmangin-fr**

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
Moved Draft PR in Cross Dev Review pipeline.
If you want someone to check, assign a reviewer.

File: src/server/rest/v1/service/ChargingStationService.ts [hunk 1]
```diff
@@ -1602,6 +1610,28 @@ export default class ChargingStationService {
...
           });
           break;
+        case Command.GET_DIAGNOSTICS:
+          result = await chargingStationClient.getDiagnostics({
+            location: params.location,
+            retries: params.retries,
+            retryInterval: params.retryInterval,
+            startTime: params.startTime,
+            stopTime: params.stopTime
+          }).catch(async (error) => {
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/ChargingStationService.ts:1620]: No, use try...catch instead

File: src/server/rest/v1/service/ChargingStationService.ts [hunk 2]
```diff
@@ -1602,6 +1610,28 @@ export default class ChargingStationService {
...
+            // Log
+            await Logging.logException(error, ServerAction.CHARGING_STATION, chargingStation.id,
+              MODULE_NAME, 'GetDiagnostics', tenant.id ?? Constants.DEFAULT_TENANT);
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/ChargingStationService.ts:1623]: No charging station on default tenant here, not a use case to handle.

File: src/server/rest/v1/service/ChargingStationService.ts (hunk 1 above)
Comment (LucasBrazi06) [reply src/server/rest/v1/service/ChargingStationService.ts:1620]: And maybe this is a global try...catch for all OCPP commands?

File: src/server/rest/v1/validator/ChargingStationValidator.ts [hunk 3]
```diff
@@ -139,6 +141,12 @@ export default class ChargingStationValidator extends SchemaValidator {
     return data;
   }
 
+  public validateChargingStationGetDiagnostics(data: HttpChargingStationGetDiagnostics): HttpChargingStationGetDiagnostics {
```
Comment (LucasBrazi06) [inline src/server/rest/v1/validator/ChargingStationValidator.ts:144]: Missing Request at the end of the HttpChargingStationGetDiagnostics like the others ;-)

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/server/rest/v1/service/ChargingStationService.ts [hunk 4]
```diff
@@ -1154,6 +1155,27 @@ export default class ChargingStationService {
     // Get the Charging station
     const chargingStation = await UtilsService.checkAndGetChargingStationAuthorization(
       req.tenant, req.user, filteredRequest.chargingStationID, action, null, { withSite: true, withSiteArea: true });
+    // Get the OCPP Client
+    const chargingStationClient = await ChargingStationClientFactory.getChargingStationClient(req.tenant, chargingStation);
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/ChargingStationService.ts:1159]: Why do you get the charging station client here?
I don't see where it is used by reading the code.

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/server/rest/v1/service/ChargingStationService.ts (hunk 4 above)
Comment (rmangin-fr) [reply src/server/rest/v1/service/ChargingStationService.ts]: Removed, this was copied from executeChargingStationCommand in a previous version :)

**Review by LucasBrazi06 — APPROVED** [review]

**Pull request merged**